src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int counter;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+        counter = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        resize();
+        values[counter] = value;
+        counter++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0) {
+            throw new ArrayIndexOutOfBoundsException("This index is out of bounds.");
+        }
+        resize();
+        System.arraycopy(values, index, values, index + 1, counter - index);
+        counter++;
+        values[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= counter || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("This index is out of bounds.");
+        }
```

Comment:
let's move this to a separate method - you use this check in get(), set() and remove()